crates/ibc/src/core/context.rs

```diff
@@ -126,7 +163,26 @@ mod val_exec_ctx {
                     }
                 }
                 .map_err(RouterError::ContextError),
-                MsgEnvelope::Channel(_message) => todo!(),
+                MsgEnvelope::Channel(message) => {
+                    let module_id = self.lookup_module(&message).map_err(ContextError::from)?;
+                    if !self.has_route(&module_id) {
+                        return Err(ChannelError::RouteNotFound)
+                            .map_err(ContextError::ChannelError)
+                            .map_err(RouterError::ContextError);
+                    }
+
+                    match message {
+                        ChannelMsg::OpenInit(_) => todo!(),
+                        ChannelMsg::OpenTry(message) => {
+                            chan_open_try_validate(self, module_id, message)
```

Comment:
I thought how about calling `chan_open_try::validate` directly instead of adding another layer of calls by `chan_open_try_validate`? (or just keeping `chan_open_try_validate`) 
Can't we merge these two somehow? The reasoning behind is not clear, making the code a bit hard to read and understand.

Reply:
The current design is based on the standard: `chan_open_try_validate()` is the validation part of the [ICS-26 router function](https://github.com/cosmos/ibc/blob/main/spec/core/ics-026-routing-module/README.md#channel-lifecycle-management) (which is currently incorrect in the spec), and `chan_open_try::validate()` is the validation part of the [core handler](https://github.com/cosmos/ibc/blob/main/spec/core/ics-004-channel-and-packet-semantics/README.md#opening-handshake).

In other words:
- `chan_open_try_validate()` calls both the core handler and the application/module callback (getting the channel id from the core handler,  and the channel version from the callback)
- `chan_open_try::validate()` is the core handler validation logic

Does this make sense? Do you still think it would be better to move away from this design? At the very least, using better naming to encode what I just explained would help.

Reply:
On this basis, `chan_open_try_validate` operates more as a handler than a validator, shouldn't contain any validation step. Also, the latter link you sent seems to correspond more with module interfaces iiuc (e.g. `on_chan_open_try_validate`).
Btw, I don't think naming help with this issue. I propose all checkings be done in one place either in `chan_open_try::validate()` or `on_chan_open_try_validate`, like what we have now for `on_chan_open_try_execute` or like before structure for `validate()` `execute()`

Reply:
Maybe not bad, keeping this change, and see what the implications will be when we refactor others. Beside this, it LGTM and your explanations are great, thanks 🙏

Reply:
Indeed I think we should leave it as is for now because correctness doesn't seem to be a concern; we will focus on improving readability/maintainability once `Validation/ExecutionContext` have shipped. WDYT?

Reply:
Agree, let's keep it for now 👍